ludwig/hyperopt/execution.py

```diff
@@ -506,7 +504,7 @@ def _checkpoint_progress(self, trainer, progress_tracker, save_path) -> None:
                 """Checkpoints the progress tracker."""
                 if is_using_ray_backend:
                     trainer_ckpt = Checkpoint.from_directory(save_path)
-                    ckpt_ref = trainer_ckpt.to_object_ref()
+                    ckpt_ref = ray.put(trainer_ckpt.to_dict(), _owner=ray_queue.actor)
```

Comment:
Or is this needed due to some reference counting issue on the Ray side (it wasn't an issue before, was it a regression?)?

Would be useful to add a comment explaining and linking to relevant Ray issue if its exists.

Reply:
This is needed because there are actually three different processes operating for a given trial– (1) the main process, which is what calls `tuner.fit()`, (2) the trial function process, which spawns the `ray_queue` object, and (3) the remote task `HorovodTrainer` creates to ultimately call our function `ludwig.backend.ray.train_fn`. The original checkpoint object `trainer_ckpt` is produced/owned by process #3, but we are trying to grab it from process #2 with `ray_queue.get_nowait_batch` here: https://github.com/ludwig-ai/ludwig/blob/402511f2db6eb2737ed1ca538533abfa85b3485e/ludwig/hyperopt/execution.py#L617

I believe this is failing because process #3 is terminated before we can get the object reference, causing an Owner Died error on the now orphaned object reference. The solution to this was using the `ray.put` call above with the `_owner` kwarg, which specifies `ray_queue.actor` as the owner of the checkpoint object reference. By doing this, we ensure the checkpoint's lifecycle is tied to `ray_queue`, which itself is tied to process #2.

I'm not entirely clear with the behavior prior to this change– perhaps process #2 and #3 used to be the same process, therefore avoiding this sort of error– @arnavgarg1 may be able to provide some context. In any case, I'll add a clarification comment.